src/eks-mcp-server/awslabs/eks_mcp_server/eks_resiliency_handler.py

```diff
@@ -0,0 +1,6069 @@
+                'check_results': [error_check_result],
+                'overall_compliant': False,
+                'summary': error_summary,
+            }
+            json_output = json.dumps(response_dict, indent=2)
+            response.content.append(
+                TextContent(
+                    type='text', text=f'Complete JSON output:\n```json\n{json_output}\n```'
+                )
+            )
+
+            return response
+
+    def _check_singleton_pods(self, k8s_api, namespace: Optional[str] = None) -> Dict[str, Any]:
+        """Check A1: Singleton pods without controller management."""
+        singleton_pods = []
+        try:
+            logger.info(
+                f'Starting singleton pods check, namespace: {namespace if namespace else "all"}'
+            )
+
+            # Prepare kwargs for filtering
+            kwargs = {}
+            if namespace:
+                kwargs['namespace'] = namespace
+                logger.info(f'Checking pods in namespace: {namespace}')
+            else:
+                logger.info('Checking pods across all namespaces')
+
+            # Use the K8sApis list_resources method
+            logger.info('Calling list_resources to get pods')
+            pods_response = k8s_api.list_resources(kind='Pod', api_version='v1', **kwargs)
+
+            # Log the number of pods found
+            pod_count = len(pods_response.items) if hasattr(pods_response, 'items') else 0
+            logger.info(f'Retrieved {pod_count} pods')
+
+            # Process the response
+            for pod in pods_response.items:
+                try:
+                    pod_dict = pod.to_dict() if hasattr(pod, 'to_dict') else pod
+
+                    # Check if pod has owner references
+                    metadata = pod_dict.get('metadata', {})
+                    owner_refs = metadata.get('ownerReferences', [])
+
+                    # Make sure we get the namespace from the pod metadata
+                    pod_namespace = metadata.get('namespace')
+                    if not pod_namespace:
+                        logger.warning("Pod missing namespace information, using 'default'")
+                        pod_namespace = 'default'
+
+                    name = metadata.get('name', 'unknown')
+
+                    logger.debug(f'Checking pod {pod_namespace}/{name}')
+
+                    if not owner_refs:
+                        logger.info(f'Found singleton pod: {pod_namespace}/{name}')
+                        singleton_pods.append(f'{pod_namespace}/{name}')
+                except Exception as pod_error:
+                    logger.error(f'Error processing pod: {str(pod_error)}')
+
+            is_compliant = len(singleton_pods) == 0
+            details = f'Found {len(singleton_pods)} singleton pods not managed by controllers'
+            logger.info(details)
+
+            remediation = """Convert singleton pods to Deployments, StatefulSets, or other controllers:
```

Comment:
Is there any way to extract some of the data for each check to a separate file, kind of like `data/eks_cloudwatch_metrics_guidance.json`? This file is quite long.